main.cpp

```diff
@@ -42,15 +45,25 @@ void PrintkFixerMLIL(Ref<AnalysisContext> analysisContext) {
       analysisContext->GetMediumLevelILFunction();
   Ref<BinaryView> bv = analysisContext->GetFunction()->GetView();
 
-  // If we don't find printk, this probably isn't a kernel module...
-  // TODO: don't re-find this every time the Binary Ninja core calls us.
-  auto sym = bv->GetSymbolByRawName("printk");
-  if (!sym) {
-    LogWarn(
-        "Failed to find printk: PrintkFixer won't do anything (is this a Linux "
-        "kernel module?)");
-    return;
+  // The scoped_lock allows us to lock all analysis threads to have only 1
+  // thread search for the symbol reference to printk.
+  {
+    std::scoped_lock<std::mutex> lock(g_initialAnalysisMutex);
```

Comment:
BN multithreads analysis, so I worry we'd end up with lock contention here. A reader-writer lock would probably work, but it's moot for the reason I've noted above

Reply:
Lock contention is intended here. The point is that we'd only have 1 thread query the bndb for the printk reference and have it shared among the other threads. If we don't then it could lock up binary ninja's analysis especially when we're regenerating the IL.